### ISSUE TITLE
IndexError when handling exceptions without arguments in python<=3.11 

### ISSUE THREAD

**Issue by thoresma** [pipefunc/pipefunc#430]
https://github.com/pipefunc/pipefunc/blob/af5e727ae765f0473cee4ee20326b5d761cfec4d/pipefunc/_utils.py#L89

```
File /local_disk0/.ephemeral_nfs/cluster_libraries/python/lib/python3.10/site-packages/pipefunc/_utils.py:86, in handle_error(e, func, kwargs)
     84 msg = f"Error occurred while executing function `{call_str}`."
     85 if sys.version_info <= (3, 11):  # pragma: no cover
---> 86     raise type(e)(e.args[0] + msg) from e
     87 e.add_note(msg)
     88 raise

IndexError: tuple index out of range
```

**Comment by basnijholt** [comment]
Thanks for reporting, should be an easy fix. Will get to it ASAP.

**Comment by basnijholt** [comment]
I just released [0.39.0](https://github.com/pipefunc/pipefunc/releases/tag/v0.39.0) that fixes this problem :tada:

Thank you for using pipefunc @thoresma! Let me know if you have any other pain points or ideas for improvements 😄
